### ISSUE TITLE
Questions about training data during iterative DPO

### ISSUE THREAD

**Issue by hong-xl** [RLHFlow/Online-RLHF#5]
Hi, awesome work and thanks for open source!

In each iteration, it seems that only using the 20k dataset annotated by reward model. However, the pipeline in the paper shows that the historical dataset will grow. Please let me know  if I have missed something important. Thanks a lot!

**Comment by WeiXiongUST** [comment]
Thanks for the nice word!

We start with pi0 (the sft model), and in the iteration 1, we annotate 20k dataset. Then, we train on the 20k data to get pi1. In the second iteration, we also annotate 20k data and **we start with pi1** to train on these new 20k data to get pi2.

In other words, instead of restart the training on the historical data 20k + 20k from pi0 (as did in Claude and our icml paper), we use the model obtained from the last iteration as the starting point of the new iteration. However, we remark that we always use pi0 as the reference model for the KL regularization. This recipe does not lead to performance regression based on our experience and should save a lot of training time.

**Comment by hong-xl** [comment]
Thanks for your patience ! 

When we evaluate RLHFlow/LLaMA3-iterative-DPO-final on mt-bench, we only get the score of 8.22. This is the code we modified from official codebase of mt-bench:

pipeline = transformers.pipeline("text-generation", model="RLHFlow/LLaMA3-iterative-DPO-final on mt-bench", model_kwargs={"torch_dtype": torch.bfloat16}, device_map="auto",)
terminators = [pipeline.tokenizer.eos_token_id, pipeline.tokenizer.convert_tokens_to_ids("<|eot_id|>")]
... 
messages = [{"role": "user", "content": query},]
outputs = pipeline(messages, max_new_tokens=1024, eos_token_id=terminators, do_sample=do_sample, temperature=temperature)

Please let me know what the difference is from your evaluation. Thanks!

**Comment by WeiXiongUST** [comment]
There is some license issue on the official checkpoint release... We are still working on this with SF and hopefully we can release them soon...
